src/fivetran-destination/src/destination/dml.rs

```diff
@@ -427,15 +392,68 @@ async fn delete_files(
     }
 
     // Copy into a temporary table, which we then merge into the destination.
+    let (qualified_temp_table_name, columns) = create_temporary_table(table, client).await?;
+    let row_count = copy_files(
+        file_config,
+        delete_files,
+        client,
+        &qualified_temp_table_name,
+    )
+    .await
+    .context("delete_files")?;
+
+    // Skip the update if there are no rows to delete!
+    if row_count == 0 {
+        return Ok(());
+    }
+
+    // Mark rows as deleted by "merging" our temporary table into the destination table.
     //
+    // HACKY: We want to update the "_fivetran_synced" column for all of the rows we marked as
+    // deleted, but don't have a way to read from the temp table that would allow this in an
+    // `UPDATE` statement.
+    let synced_time_stmt = format!(
+        "SELECT MAX({synced_col}) FROM {qualified_temp_table_name}",
+        synced_col = escape::escape_identifier(FIVETRAN_SYSTEM_COLUMN_SYNCED)
+    );
+    let synced_time: SystemTime = client
+        .query_one(&synced_time_stmt, &[])
+        .await
+        .and_then(|row| row.try_get(0))?;
+
+    let qualified_table_name =
+        UnresolvedItemName::qualified(&[Ident::new(schema)?, Ident::new(&table.name)?]);
+    let matching_cols = columns.iter().filter(|col| col.is_primary);
+    let merge_stmt = format!(
+        r#"
+        UPDATE {qualified_table_name}
+        SET {deleted_col} = true, {synced_col} = $1
+        WHERE ({cols}) IN (
+            SELECT {cols}
+            FROM {qualified_temp_table_name}
+        )"#,
+        deleted_col = escape::escape_identifier(FIVETRAN_SYSTEM_COLUMN_DELETE),
+        synced_col = escape::escape_identifier(FIVETRAN_SYSTEM_COLUMN_SYNCED),
+        cols = matching_cols.map(|col| &col.name).join(","),
+    );
+    let total_count = client.execute(&merge_stmt, &[&synced_time]).await?;
+    tracing::info!(?total_count, "altered rows in {qualified_table_name}");
+
+    Ok(())
```

Comment:
this is pure code movement, I pulled creating a temporary table into a separate method